Add the Actions menu to Observation Lists when seen in a Project context

mo-nathan opened the pull request:
I think it now makes sense to make these actions available in the Project context.  I removed them originally due to all the clutter.

However, I've also realized that the Project context version of this page now needs the species list pager that I apparently neglected back when I added the Project context, `add_pager_for(@species_list)`.  Just adding that code in `app/views/controllers/species_lists/show.html.erb` doesn't seem to work.

mo-nathan committed:
Add the Actions menu to Observation Lists when seen in a Project context

nimmolo committed:
Rearrange logic for banner and title

nimmolo committed:
Rearrange project tabs and page title for species list

nimmolo committed:
Don't show title again on projects with banner

nimmolo committed:
Fix tests

nimmolo commented:
What i'm doing is 
- changing add_project_banner from assigning content_for(:title) to content_for(:banner_title)
- changing the project show page to also add_page_title for the html title element, but not print it on the page if there is a project banner. 
- moving the order of the header so that the #title (e.g. for a list) may print below the project banner and project tabs. 
- printing a title on show species_list for project and a few other alias pages where it was missing.

nimmolo committed:
Update title_prev_next_helper.rb

nimmolo committed:
Add check_param to QueryRecord

nimmolo committed:
Use QueryRecord.check_param to set the @project ivar on species_list show

nimmolo committed:
Use link_with_query for individual species_list links (project is within query)

nimmolo committed:
Refactor page_title to always show edit icons, interest icons and pager

nimmolo committed:
On projects, assign id #title to the banner title. (Others get #banner_title)

nimmolo committed:
Use `set_project_ivar` also in species list index

So that when navigating back to the index from a project SpL, you get the banner

nimmolo committed:
fix set_project_ivar

nimmolo committed:
Only show filters if no banner

nimmolo committed:
Fix conditional filter layout on projects

nimmolo committed:
Update _index_bar.erb

nimmolo committed:
Add index pagination to species list show, in addition to show pager next/prev

nimmolo committed:
Update show.html.erb

nimmolo committed:
Fix confusing layout of species list id

Seems to be the number of obs, but it's the id. Use CSS "badge" to make it clear.

nimmolo committed:
Update _page_title.erb

Fix logic for projects pages other than show/index

mo-nathan committed:
Simplify the title logic

mo-nathan committed:
Add test for presence of title text

nimmolo committed:
Update title_interest_icons_helper.rb
